### ISSUE TITLE
Support for adding light gateways

### ISSUE THREAD

**Issue by Vagabond** [helium/blockchain-core#735]
Now most of the groundwork has been laid for light gateways, it's time to add a way to add them to the chain.

For simplicity the plan is to re-use the existing add_gateway txn but if the 'payer' is not one of the accounts blessed in the `staking_keys` chain variable we will add the gateway as a light gateway. Light gateways will only cost $20 to onboard, not $40 and will only be able to, initially, forward device packets.

A chain var will be added to allow the capabilities of light gateways to be ratcheted forwards as we build out more functionality (witnessing/being challenged, etc).

- [ ] Add field to ledger gateway to track if a gateway is full or light
- [ ] Check payer against staking_keys for add_gateway txn and flag gateway as light and lower fee if the payer is not a staking key
- [ ] Check the gateway type is full for challenging/being challenged/witnessing/etc
- [ ] Add chain var we can check for enabling further light client capabilities (could be a bitmask)

**Comment by abhay** [comment]
We may have to remember to allow the [old](https://explorer.helium.com/accounts/14fzfjFcHpDR1rTH8BNPvSi5dKBbgxaDnmsVPbCjuq9ENjpZbxh) `staking_key` in the permitted list.

**Comment by Vagabond** [comment]
No, because that was valid in the ledger at the time, but it no longer is.

> We may have to remember to allow the [old](https://explorer.helium.com/accounts/14fzfjFcHpDR1rTH8BNPvSi5dKBbgxaDnmsVPbCjuq9ENjpZbxh) `staking_key` in the permitted list.

No, that was valid at the time it was valid, but it's been revoked so new transactions do not need to honor it.

**Comment by abhay** [comment]
> No, that was valid at the time it was valid, but it's been revoked so new transactions do not need to honor it.

Ah, so we'll just grandfather everyone in for now. SG

**Comment by Vagabond** [comment]
Yes, all existing gateways should be designated as full.

**Comment by bitlite** [comment]
> Will these light-gateways from approved Makers be eligible for earning for proof-of-coverage once this is supported?

That is the ticket as per the definition of light hotspots.  Light hotspots will not include any P2P systems and therefore will work with the Validators for POC.


> Previously my understanding was that PoC-ing light-gateways would have to pay the full 40 USD fee and only never PoC-ing light-gateways would have to pay the 20 USD fee, but in this case I see no reason to onboard non-PoC-ing light-gateways to the onboarding server as a Maker, because non-onboarded DIY-like light-gateways will earn the same.

There will be 3 types of units.
- Full Gateways (currently in the wild)
- Light Gateways (coming Q1 2022)
- Data-Only Gateways (does not make any HNT or may only make on Transfers *To be clarified*)
